src/Microsoft.ML/Runtime/Internal/Tools/CSharpApiGenerator.cs

```diff
@@ -798,10 +850,15 @@ string GetFriendlyTypeName(string currentNameSpace, string typeName)
             foreach (var line in entryPointInfo.Description.Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries))
                 writer.WriteLine($"/// {line}");
             writer.WriteLine("/// </summary>");
-            writer.WriteLine($"public sealed partial class {classAndMethod.Item2}{classBase}");
+
+            string seal = entryPointInfo.NoSeal ? "" : "sealed ";
+            writer.WriteLine($"public {seal}partial class {classAndMethod.Item2}{classBase}");
             writer.WriteLine("{");
             writer.Indent();
             writer.WriteLine();
+            if (classBase.Contains("ILearningPipelineLoader"))
```

Comment:
>ILearningPipelineLoader [](start = 36, length = 23)

So I'm curious, I might have expected these "special" kinds like these to be detected using direct type comparisons, on `entryPointInfo.InputKinds`. That code would be considerably safer and simpler than searching strings, but does something make that impossible?

Having these detections be string comparisons makes me uneasy. This makes the usage invisible to any refactoring utilities (either within VS or Resharper), so from that perspective even a `nameof` would make me feel better. But this `.Contains` strategy also will catch things where the name is either a prefix or suffix of a type. (E.g., testing against `IFoo` will match `IFooBar`, which is not intentional.)